src/linalg/impl_linalg.rs

```diff
@@ -532,6 +595,35 @@ fn blas_compat_1d<A, S>(a: &ArrayBase<S, Ix1>) -> bool
 }
 
 #[cfg(feature="blas")]
+fn blas_compat<A, S, D>(a: &ArrayBase<S, D>) -> bool
+    where S: Data,
+          A: 'static,
+          S::Elem: 'static,
+          D: Dimension,
+{
+    if !same_type::<A, S::Elem>() {
+        return false;
+    }
+
+    if a.len() > blas_index::max_value() as usize {
+        return false;
+    }
```

Comment:
Looking at [`saxpy.f`](http://www.netlib.org/lapack/explore-html/df/d28/group__single__blas__level1_gad2a52de0e32a6fc111931ece9b39726c.html#gad2a52de0e32a6fc111931ece9b39726c):

```fortran
DO i = 1,n
    sy(iy) = sy(iy) + sa*sx(ix)
    ix = ix + incx
    iy = iy + incy
END DO
```

It looks like we need `a.len() * stride`?